gaeguli/pipeline.c

```diff
@@ -586,15 +579,31 @@ _get_source_description (GaeguliPipeline * self)
   return g_string_free (result, FALSE);
 }
 
+static gchar *
+_get_stream_props_description (GaeguliPipeline * self)
+{
+  GString *stream_props = g_string_new (PIPEWIRE_NODE_STREAM_PROPERTIES_STR);
+
+  g_string_append_printf (stream_props, ",%s=%p_%d",
+      GAEGULI_PIPELINE_TAG, self, getpid ());
+
+  return g_string_free (stream_props, FALSE);
+}
+
 static gchar *
 _get_vsrc_pipeline_string (GaeguliPipeline * self)
 {
   g_autofree gchar *source = _get_source_description (self);
+  g_autofree gchar *props = _get_stream_props_description (self);
 
+  /* FIXME - pipewiresink along with another sink in the pipeline   *
```

Comment:
What is the problem here exactly? Why doesn't the pipeline with two sinks play?

Reply:
The consumer pipeline makes the provider pipeline crash. When pipewiresrc tries to get the captured video, it seems that the buffers provided to pipewiresink in provider pipeline are NULL and the pipeline crashes. 

Crash is seen when v4l2src is used. When pipewiresrc is used in the provider pipeline, pipeline won't crash but the buffers provided to pipewiresink seem to be of unknown data and eventually no data is received by the consumer pipeline.

This can be reproduced with the below pipelines.
gst-launch-1.0 v4l2src device=/dev/video0 ! image/jpeg,height=720,width=1280,framerate=30/1 !         decodebin name=decodebin ! clockoverlay name=overlay  ! tee name=tee allow-not-linked=1  !         queue ! pipewiresink mode=provide tee. ! queue ! valve name=valve drop=0 ! jpegenc name=jpegenc ! jifmux name=jifmux ! fakesink name=fakesink async=0

gst-launch-1.0 pipewiresrc  ! videoconvert ! ximagesink

Reply:
Alright, I'll try to look into this on Monday while you can focus on the next step in multiprocessing: management of the child processes.